lib/Specio/Exception.pm

```diff
@@ -50,6 +53,12 @@ sub as_string {
     my $self = shift;
 
     my $str = $self->message;
+    if ($self->context and my %context = %{$self->context}) {
+        $str
+          .= ' ('
+          . join(', ', map { "$_: $context{$_}" } sort keys %context)
```

Comment:
We could just use `Specio::PartialDump` here, which would make even more sense if this wasn't required to be a hashref.